downloader/src/main/kotlin/vcs/GitWorkingTree.kt

```diff
@@ -39,6 +41,19 @@ open class GitWorkingTree(workingDir: File, private val gitBase: GitBase) : Work
         return File(dotGitDir, "shallow").isFile
     }
 
+    override fun getNested(): Map<String, VcsInfo> {
+        val root = getRootPath()
+
+        val paths = gitBase.run(root, "submodule", "status", "--recursive").stdout.lines()
+                .filter { it.isNotBlank() }
+                .map { it.removePrefix("+").removePrefix("-").trim().split(" ")[1] }
```

Comment:
Alternatively, use `git submodule --quiet foreach --recursive "echo $sm_path"` (interestingly, the latter variable expansion also works from Windows `cmd`).

Reply:
Done.

Reply:
FYI, I had to use `$displaypath` because it is relative to the root Git repository, `$sm_path` is relative to the Git repository where the submodule is defined, so for recursive submodules this is not what we want.